tsdb/batcher.go

```diff
@@ -117,8 +117,12 @@ func (b *PointBatcher) Stop() {
 		return
 	}
 
-	close(b.stop)
-	b.wg.Wait()
+	// If not stopped, stop the batching go routine gracefully
+	if b.stop != nil {
+		b.stop <- struct{}{}
```

Comment:
That is true. Tested with some sample code in Go playground. I didn't know that a closed channel always return the default value and will always be chosen in a select. There is something new to learn everyday in this language!

Thanks!